test/WorkerExtensionTests/Table/TableConverterTests.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Google.Protobuf;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Grpc.Messages;
+using Microsoft.Azure.Functions.Worker.Tests.Converters;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+using Microsoft.Azure.Functions.Worker.Extensions.Tables.Config;
+using Microsoft.Azure.Functions.Worker.Extensions.Tables;
+using Azure.Data.Tables;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.WorkerExtension.Tests.Table
+{
+    public class TableConverterTests
+    {
+        private Mock<TableConverter> _mockTableConverter;
```

Comment:
We should write the unit tests without mocking the converter; instead mock the external source (the table client). We have this done in the cosmos db converter tests so I would reuse the work in that (we have a GH issue to update the blob tests to be the same way).

In general, we shouldn't be making things internal just to test them, you should be able to test all of the code paths via the main entry point (in this case, ConvertAsync).

https://github.com/Azure/azure-functions-dotnet-worker/blob/feature/sdk-type-binding/test/WorkerExtensionTests/Cosmos/CosmosDBConverterTests.cs

Reply:
Done!